Source/Core/Core/TimePlayed.cpp

```diff
@@ -4,39 +4,66 @@
 #include "Core/TimePlayed.h"
 
 #include <chrono>
+#include <functional>
+#include <mutex>
 #include <string>
+#include <vector>
 
 #include "Common/CommonTypes.h"
 #include "Common/FileUtil.h"
 #include "Common/IniFile.h"
 #include "Common/NandPaths.h"
 
-TimePlayed::TimePlayed() : m_ini_path(File::GetUserPath(D_CONFIG_IDX) + "TimePlayed.ini")
+TimePlayedManager::TimePlayedManager()
+    : m_ini_path(File::GetUserPath(D_CONFIG_IDX) + "TimePlayed.ini")
 {
-  Reload();
+  m_ini.Load(m_ini_path);
+  m_time_list = m_ini.GetOrCreateSection("TimePlayed");
 }
 
-TimePlayed::~TimePlayed() = default;
+TimePlayedManager::~TimePlayedManager() = default;
+
+TimePlayedManager& TimePlayedManager::GetInstance()
+{
+  static TimePlayedManager time_played_manager;
+  return time_played_manager;
+}
 
-void TimePlayed::AddTime(const std::string& game_id, std::chrono::milliseconds time_emulated)
+void TimePlayedManager::AddTime(const std::string& game_id,
+                                const std::chrono::milliseconds time_emulated)
 {
-  std::string filtered_game_id = Common::EscapeFileName(game_id);
+  const std::string filtered_game_id = Common::EscapeFileName(game_id);
   u64 previous_time;
-  m_time_list->Get(filtered_game_id, &previous_time);
-  m_time_list->Set(filtered_game_id, previous_time + static_cast<u64>(time_emulated.count()));
-  m_ini.Save(m_ini_path);
+
+  {
+    std::lock_guard guard(m_mutex);
```

Comment:
This does have proper synchronization, but it should be added in the first commit rather than the second commit.